### ISSUE TITLE
Support Route53 alias records

### ISSUE THREAD

**Issue by 2rs2ts** [spulec/moto#1599]
Please add support for alias records. You can see in the route53/responses.py that there is code reading upserted records for their AliasTarget and setting them to be the record set's `ResourceRecords` list:

https://github.com/spulec/moto/blob/f18259d49e5e66929a793ec8105d32ccfcc70d21/moto/route53/responses.py#L135-L137

I believe this is wrong; you should maintain the alias as an alias. I think normally you can't create an alias and also give it `ResourceRecords` (for example, [the CloudFormation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-route53-recordset.html#cfn-route53-recordset-resourcerecords) says not to specify it if you specify an `AliasTarget`.)

**Comment by spulec** [comment]
Thanks for opening; I'm going to set this as a feature request.

**Comment by elliottmb** [comment]
I've just run into this issue when writing tests for an application that manipulates alias targets. 

What I expect to see
```
{
    "AliasTarget": {
        "HostedZoneId": "FAKEZONE", 
        "EvaluateTargetHealth": false, 
        "DNSName": "other.example.com."
    }, 
    "Type": "A", 
    "Name": "test.example.com."
}
```

and what moto is responding with:

```
{
    'Name': 'test.example.com',
    'Type': 'A',
    'ResourceRecords': [
        {
            'Value': 'other.example.com.'
        }
    ]
}
```

This has meant that part of my code can't be tested using moto.

I can open an PR to try to fix this, if that would be the best route.

**Comment by terrycain** [comment]
@elliottmb Sure open up a PR and I'll review it.

**Comment by bblommers** [comment]
As far as I can tell this was fixed by #2082, so I'll close this. Please let us know if there's still anything wrong/outstanding though.
